### ISSUE TITLE
The model performance is greatly affected by long-tailed dataset

### ISSUE THREAD

**Issue by earthpimp** [gaozhihan/PreDiff#31]
Sorry to bother you.
I trained Prediff on my own dataset and found the result quite bad. I guess the reason behind should be data imbalance which is commonly observed in precipitation nowcasting.
I noticed that in your previous paper regarding TrajGRU, pixelwise loss weighting is applied to the radar sequence. How could I implement a similar approach in Prediff.
I would be appreciated if you could offer some suggestions.

**Comment by gaozhihan** [comment]
Thank you for your interest in our work and your question. Yes, we encountered a similar challenge when training PreDiff on highly imbalanced data, such as the [HKO-7](https://github.com/sxjscience/HKO-7?tab=readme-ov-file#download-the-hko-7-dataset-and-use-the-iterator). Although it is not feasible to directly implement a loss function like in TrajGRU that can intuitively balance the training towards data in a long-tail distribution, we found that adjusting the data sampling directly can help alleviate this problem. Specifically, we increase the sampling of rare data and decrease the sampling of common data.

**Comment by TienMinhHoa** [comment]
How many samples do you have on training set? My train set contains about 4000 samples and almost all of them contain many points of rainfall. But the result is bad. I tested many setting in VAE and LDM but the results are still bad, the images generated contain too much, What happens with me?

> Thank you for your interest in our work and your question. Yes, we encountered a similar challenge when training PreDiff on highly imbalanced data, such as the [HKO-7](https://github.com/sxjscience/HKO-7?tab=readme-ov-file#download-the-hko-7-dataset-and-use-the-iterator). Although it is not feasible to directly implement a loss function like in TrajGRU that can intuitively balance the training towards data in a long-tail distribution, we found that adjusting the data sampling directly can help alleviate this problem. Specifically, we increase the sampling of rare data and decrease the sampling of common data.
